exchanges/kraken/kraken.go

```diff
@@ -102,62 +94,50 @@ func (k *Kraken) GetAssetPairs(ctx context.Context, assetPairs []string, info st
 	}
-	var response struct {
-		Error  []string               `json:"error"`
-		Result map[string]*AssetPairs `json:"result"`
-	}
+
+	var result map[string]*AssetPairs
 	if info != "" {
 		if info != "margin" && info != "leverage" && info != "fees" && info != "info" {
-			return response.Result, errors.New("parameter info can only be 'asset', 'margin', 'fees' or 'leverage'")
+			return nil, errors.New("parameter info can only be 'asset', 'margin', 'fees' or 'leverage'")
 		}
 		params.Set("info", info)
 	}
-	if err := k.SendHTTPRequest(ctx, exchange.RestSpot, path+params.Encode(), &response); err != nil {
-		return response.Result, err
+	if err := k.SendHTTPRequest(ctx, exchange.RestSpot, path+params.Encode(), &result); err != nil {
+		return nil, err
 	}
-	return response.Result, GetError(response.Error)
+	return result, nil
 }
 
 // GetTicker returns ticker information from kraken
-func (k *Kraken) GetTicker(ctx context.Context, symbol currency.Pair) (Ticker, error) {
-	tick := Ticker{}
+func (k *Kraken) GetTicker(ctx context.Context, symbol currency.Pair) (*Ticker, error) {
 	values := url.Values{}
 	symbolValue, err := k.FormatSymbol(symbol, asset.Spot)
 	if err != nil {
-		return tick, err
+		return nil, err
 	}
 	values.Set("pair", symbolValue)
 
-	type Response struct {
-		Error []interface{}             `json:"error"`
-		Data  map[string]TickerResponse `json:"result"`
-	}
-
-	resp := Response{}
+	var data map[string]TickerResponse
 	path := fmt.Sprintf("/%s/public/%s?%s", krakenAPIVersion, krakenTicker, values.Encode())
+	if err := k.SendHTTPRequest(ctx, exchange.RestSpot, path, &data); err != nil {
+		return nil, err
+	}
 
-	err = k.SendHTTPRequest(ctx, exchange.RestSpot, path, &resp)
-	if err != nil {
-		return tick, err
-	}
-
-	if len(resp.Error) > 0 {
-		return tick, fmt.Errorf("%s error: %s", k.Name, resp.Error)
-	}
-	for i := range resp.Data {
-		tick.Ask = resp.Data[i].Ask[0].Float64()
-		tick.AskSize = resp.Data[i].Ask[2].Float64()
-		tick.Bid = resp.Data[i].Bid[0].Float64()
-		tick.BidSize = resp.Data[i].Bid[2].Float64()
-		tick.Last = resp.Data[i].Last[0].Float64()
-		tick.Volume = resp.Data[i].Volume[1].Float64()
-		tick.VolumeWeightedAveragePrice = resp.Data[i].VolumeWeightedAveragePrice[1].Float64()
-		tick.Trades = resp.Data[i].Trades[1]
-		tick.Low = resp.Data[i].Low[1].Float64()
-		tick.High = resp.Data[i].High[1].Float64()
-		tick.Open = resp.Data[i].Open.Float64()
-	}
-	return tick, nil
+	var tick Ticker
+	for i := range data {
+		tick.Ask = data[i].Ask[0].Float64()
+		tick.AskSize = data[i].Ask[2].Float64()
+		tick.Bid = data[i].Bid[0].Float64()
+		tick.BidSize = data[i].Bid[2].Float64()
+		tick.Last = data[i].Last[0].Float64()
+		tick.Volume = data[i].Volume[1].Float64()
+		tick.VolumeWeightedAveragePrice = data[i].VolumeWeightedAveragePrice[1].Float64()
+		tick.Trades = data[i].Trades[1]
+		tick.Low = data[i].Low[1].Float64()
+		tick.High = data[i].High[1].Float64()
+		tick.Open = data[i].Open.Float64()
+	}
```

Comment:
```suggestion
	for _, v := range data {
		tick.Ask = v.Ask[0].Float64()
		tick.AskSize = v.Ask[2].Float64()
		tick.Bid = v.Bid[0].Float64()
		tick.BidSize = v.Bid[2].Float64()
		tick.Last = v.Last[0].Float64()
		tick.Volume = v.Volume[1].Float64()
		tick.VolumeWeightedAveragePrice = v.VolumeWeightedAveragePrice[1].Float64()
		tick.Trades = v.Trades[1]
		tick.Low = v.Low[1].Float64()
		tick.High = v.High[1].Float64()
		tick.Open = v.Open.Float64()
	}
```

Limit map lookup and use struct instead and if you can update the ones below can optimise those as well.